JoyboyCommunity/src/screens/Auth/ImportKeys.tsx

```diff
@@ -0,0 +1,114 @@
+import {NDKPrivateKeySigner} from '@nostr-dev-kit/ndk';
+import {canUseBiometricAuthentication} from 'expo-secure-store';
+import {useState} from 'react';
+import {Platform} from 'react-native';
+
+import {LockIcon} from '../../assets/icons';
+import {Button, Input} from '../../components';
+import {useNostrContext} from '../../context/NostrContext';
+import {useTheme} from '../../hooks';
+import {useDialog, useToast} from '../../hooks/modals';
+import {Auth} from '../../modules/Auth';
+import {AuthImportKeysScreenProps} from '../../types';
+import {isValidNostrPrivateKey} from '../../utils/keypair';
+import {storePassword, storePrivateKey, storePublicKey} from '../../utils/storage';
+
+export const ImportKeys: React.FC<AuthImportKeysScreenProps> = ({navigation}) => {
+  const {theme} = useTheme();
+
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [privateKey, setPrivateKey] = useState('');
+  const [publicKey, setPublicKey] = useState('');
+
+  const {ndk} = useNostrContext();
+  const {showToast} = useToast();
+  const {showDialog, hideDialog} = useDialog();
+
+  const handleImportAccount = async () => {
+    if (!username) {
+      showToast({type: 'error', title: 'Username is required'});
+      return;
+    }
+
+    if (!password) {
+      showToast({type: 'error', title: 'Password is required'});
+      return;
+    }
+
+    if (!privateKey) {
+      showToast({type: 'error', title: 'Private key to import is required'});
+      return;
+    }
+
+    if (!isValidNostrPrivateKey(privateKey)) {
+      showToast({type: 'error', title: 'Private key not valid'});
+      return;
+    }
+    await storePrivateKey(privateKey, password);
+    ndk.signer = new NDKPrivateKeySigner(privateKey);
+    const user = await ndk.signer.user();
+    const publicKey = user?.pubkey;
+    setPublicKey(publicKey);
+    const ndkUser = ndk.getUser({pubkey: user?.pubkey});
+    ndkUser.profile = {nip05: username};
+    await storePublicKey(publicKey);
+    await ndkUser.publish();
+
+    const biometySupported = Platform.OS !== 'web' && canUseBiometricAuthentication();
+    if (biometySupported) {
+      showDialog({
+        title: 'Easy login',
+        description: 'Would you like to use biometrics to login?',
+        buttons: [
+          {
+            type: 'primary',
+            label: 'Yes',
+            onPress: async () => {
+              await storePassword(password);
+              hideDialog();
+            },
+          },
+          {
+            type: 'default',
+            label: 'No',
+            onPress: hideDialog,
+          },
+        ],
+      });
+    }
+
+    navigation.navigate('SaveKeys', {privateKey, publicKey});
+  };
+
+  return (
+    <Auth title="ImportKeys">
+      <Input placeholder="@ Username" value={username} onChangeText={setUsername} />
```

Comment:
Username should be removed, since importing means the account is already setup.

Reply:
I do it because the user can have a Nostr private key, without a username and NIP-5 setup on the Joyboy relayer.
Ok let's not enabled set it on the first time and do an edit profile if needed